### DISCUSSION TITLE
Can I prevent cache generation and saving if it matches the restore key?

### DISCUSSION THREAD

**Discussion by hkusu** [actions/cache#890] [Q&A]
Cache is generated even when the restore key is applied. 

There is a cache capacity limit (10G), so in that case I don't want to generate a cache, but is there any way?

It would also be nice to have a feature like read-only that doesn't generate cache even if it doesn't match the restore key.

**Comment by henrygab** [comment]
Yes, please!!!

Here's an actual use case:

1. scheduled job runs `actions/cache` with a fixed, never-to-be-used key, and with restore-keys listing only the common prefix (e.g., force miss + use of restore key to get latest version)
1. workflow runs install / update / upgrade items
1. workflow generate a new hash, based on the new files
1. if new hash matches hard-coded expected hash:
    **_workflow exits successfully_** (**_but without creating a new cache!!!_**)
1. workflow runs `actions/cache` a **second** time, now using using the prefix + new hash ... with the expectation that this would create the cache using the prefix + new hash
1. workflow opens an issue (later: maybe a full PR) in the repository, indicating the new hash that needs to be used in the affected workflow(s)

Basically, there is a need to disable automatic cache creation.  Put another way, there needs to be an option to run the action in a "read-only" mode, where all it does is attempt to restore a cache, if a match can be found.

Without this ability to prevent cache generation, the cache system breaks for many uses.
